### PR TITLE
Refactor callstack_spec using compile and run helper

### PR DESCRIPTION
**Pull request by straight-shoota**
This PR simply extracts the behaviour to compile a crystal file and run the resulting executable into a reusable helper method.

### CODE REVIEW

File: spec/std/callstack_spec.cr [hunk 1]
```diff
@@ -1,68 +1,62 @@
...
-    with_tempfile("compiler_spec_output") do |path|
-      sample = datapath("backtrace_sample")
+private def compile_and_run_file(source_file)
+  with_tempfile("executable_file") do |executable_file|
+    puts executable_file
+    Process.run("bin/crystal", ["build", "--debug", "-o", executable_file, source_file])
+    File.exists?(executable_file).should be_true
+
+    {% if flag?(:darwin) %}
+      Process.run("dsymutil", ["--flat", executable_file])
```
Comment (asterite) [inline spec/std/callstack_spec.cr]: Now that I think about it, this might not be needed anymore because IIRC we always run dsymutil in mac now

Comment (straight-shoota) [reply spec/std/callstack_spec.cr]: I was not sure about this either, but don't have a mac for testing. I'll remove it.

File: spec/std/callstack_spec.cr [hunk 2]
```diff
@@ -1,68 +1,58 @@
 require "./spec_helper"
 
-describe "Backtrace" do
-  it "prints file line:colunm" do
-    with_tempfile("compiler_spec_output") do |path|
-      sample = datapath("backtrace_sample")
+private def compile_and_run_file(source_file)
+  with_tempfile("executable_file") do |executable_file|
+    puts executable_file
```
Comment (RX14) [inline spec/std/callstack_spec.cr]: This debug code shouldn't have been merged.

Comment (straight-shoota) [reply spec/std/callstack_spec.cr]: Ooops.

**Pull request merged**